core/drivers/dcp/imx_dcp_utils.c

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020 NXP
+ */
+#include <drivers/dcp/imx_dcp.h>
+#include <imx_dcp_utils.h>
+#include <malloc.h>
+#include <mm/core_memprot.h>
+#include <trace.h>
+
+struct __packed mem_hdr {
+	void *addr;
+	size_t size;
+};
+
+#define MEM_HDR_SIZE  ROUNDUP(sizeof(struct mem_hdr), sizeof(void *))
+#define OF_MEM_HDR(p) (struct mem_hdr *)((uint8_t *)(p) - (MEM_HDR_SIZE))
+
+/*
+ * Read the system cache line size.
+ * Get the value from the ARM system configuration register
+ */
+static uint32_t read_cacheline_size(void)
```

Comment:
factorize with `read_cacheline_size()` from **core/drivers/crypto/caam/utils/utils_mem.c**?

Reply:
Should we move this function in the core to make it available to all drivers?